src/kem.rs

```diff
@@ -78,7 +85,18 @@ pub fn key_gen(
 ) -> Result<(Vec<u8>, Vec<u8>), Error> {
     match alg {
         Algorithm::X25519 | Algorithm::Secp256r1 => {
-            ecdh::key_gen(alg.into(), rng).map_err(|e| e.into())
+            ecdh::key_gen(alg.try_into().unwrap(), rng).map_err(|e| e.into())
+        }
+
+        Algorithm::Kyber768 => {
+            let mut seed = [0; kyber768::KEY_GENERATION_SEED_SIZE];
+            rng.try_fill_bytes(&mut seed).map_err(|_| Error::KeyGen)?;
```

Comment:
We have the same in `encapsulate`. Maybe this should be moved out into a function.